Uplift.Api/Controllers/CategoryController.cs

```diff
@@ -15,18 +15,72 @@ public CategoryController(ICategoryService categoryService)
             _categoryService = categoryService;
         }
 
-        [HttpPost("create-category")]
-        public IActionResult Create([FromBody] Category model)
+        // GET: api/Category
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Category>>> GetCategories()
         {
-            _categoryService.Create(model);
-            return Ok();
+            var categories = await _categoryService.GetAllCategoriesAsync();
+            return Ok(categories);
         }
 
-        [HttpPost("Get-category")]
-        public ActionResult<IEnumerable<Category>> GetCategories()
+        // GET: api/Category/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Category>> GetCategory(int id)
         {
-            var categories = _categoryService.GetAllCategories();
-            return Ok(categories);
+            var category = await _categoryService.GetCategoryByIdAsync(id);
+            if (category == null)
+            {
+                return NotFound();
+            }
+            return Ok(category);
+        }
+
+        // POST: api/Category
+        [HttpPost]
+        public async Task<ActionResult<Category>> CreateCategory([FromBody]Category category)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
```

Comment:
That is good u use relevant responses to the client in order to handle the errors.

Reply:
Thanks. I need to read more about this.